components/FlexibleHeader/src/MDCFlexibleHeaderView.m

```diff
@@ -551,65 +543,36 @@ - (MDCFlexibleHeaderScrollViewInfo *)fhv_addInsetsToScrollView:(UIScrollView *)s
-      if (scrollView.contentInset.top != scrollView.adjustedContentInset.top) {
-        // Only adjust for safe area insets if the content inset of it is affected by it.
-        safeAreaAdjustment = MDCDeviceTopSafeAreaInset();
-      }
-    }
+  CGFloat existingContentInsetAdjustment =
+      [self fhv_existingContentInsetAdjustmentForScrollView:scrollView];
+  CGFloat desiredTopInset = self.computedMaximumHeight - existingContentInsetAdjustment;
+
+  // During modal presentation on non-X devices our top safe area inset can be much larger than it
+  // actually will be, causing desiredTopInset to be small or even negative. To guard against this,
+  // we ensure that our desired top inset is always at least the header height.
+  CGFloat minimumTopInset;
+  if (_minMaxHeightIncludesSafeArea) {
+    minimumTopInset = _maximumHeight - MDCDeviceTopSafeAreaInset();
+  } else {
+    minimumTopInset = _maximumHeight;
   }
-#endif
+  desiredTopInset = MAX(minimumTopInset, desiredTopInset);
 
+  UIEdgeInsets insets = scrollView.contentInset;
+  CGFloat topInsetAdjustment = 0;
   if (!info.hasInjectedTopContentInset) {
-    UIEdgeInsets insets = scrollView.contentInset;
-    insets.top += self.computedMaximumHeight - safeAreaAdjustment;
-    info.injectedTopContentInset = self.computedMaximumHeight - safeAreaAdjustment;
-    info.hasInjectedTopContentInset = YES;
-    info.topSafeAreaInsetAdjustment = safeAreaAdjustment;
-    info.hasTopSafeAreaInsetAdjustment = YES;
+    topInsetAdjustment = desiredTopInset;
+  } else {
+    topInsetAdjustment = desiredTopInset - info.injectedTopContentInset;
+  }
+  insets.top += topInsetAdjustment;
+  info.injectedTopContentInset = desiredTopInset;
+  info.hasInjectedTopContentInset = YES;
+  if (!UIEdgeInsetsEqualToEdgeInsets(scrollView.contentInset, insets)) {
     scrollView.contentInset = insets;
   }
 
-  // The scroll indicator insets are updated by fhv_accumulatorDidChange and change dynamically with
-  // the header.
-  return info;
-}
-
-- (void)fhv_adjustInsetsForSafeAreaInScrollView:(UIScrollView *)scrollView {
-  if (!scrollView) {
-    return;
-  }
-#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
-  if (@available(iOS 11.0, *)) {
-    if (scrollView.contentInsetAdjustmentBehavior != UIScrollViewContentInsetAdjustmentNever) {
-
-      // For Automatic: If contentInset and adjustedContentInset are the same, no-op.
-      if (scrollView.contentInset.top == scrollView.adjustedContentInset.top) {
-        return;
-      }
-
-      // If the injected Safe Area inset hasn't changed, no-op.
-      MDCFlexibleHeaderScrollViewInfo *info = [_trackedScrollViews objectForKey:scrollView];
-      if (!info || (info.hasTopSafeAreaInsetAdjustment &&
-                    info.topSafeAreaInsetAdjustment == MDCDeviceTopSafeAreaInset())) {
-        return;
-      }
-
-      // We update the min and max height if we're still using the defaults.
```

Comment:
As discussed over chat, I think we still need this. This helps us catch any changes to the safe area inset that we might've missed for things like [UIApplication mdc_safeSharedApplication].keyWindow not being set when safeAreaInsetsDidChange gets called.

Reply:
Added back.